src/geometry/manifold/derivation_bundle.lean

```diff
@@ -0,0 +1,152 @@
+/-
+Copyright © 2020 Nicolò Cavalleri. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Nicolò Cavalleri
+-/
+
+import geometry.manifold.algebra.smooth_functions
+import ring_theory.derivation
+
+/-!
+
+# Derivation bundle
+
+In this file we define the derivations at a point of a manifold on the algebra of smooth fuctions.
+Moreover we define the total bundle of derivations (although at the moment it has not been given a
+topology). Finally we define the differential of a function in terms of derivations.
+
+The content of this file is not meant to be regarded as an alternative definition to the current
+tangent bundle but rather as a purely algebraic theory that provides a purely algebraic definition
+of the Lie algebra for a Lie group.
+
+-/
+
+variables (𝕜 : Type*) [nondiscrete_normed_field 𝕜]
+{E : Type*} [normed_group E] [normed_space 𝕜 E]
+{H : Type*} [topological_space H] (I : model_with_corners 𝕜 E H)
+(M : Type*) [topological_space M] [charted_space H M] (n : with_top ℕ)
+
+open_locale manifold
+
+instance smooth_functions_algebra : algebra 𝕜 C^∞⟮I, M; 𝕜⟯ := by apply_instance
+instance smooth_functions_tower : is_scalar_tower 𝕜 C^∞⟮I, M; 𝕜⟯ C^∞⟮I, M; 𝕜⟯ := by apply_instance
+
+/-- Type synonym, introduced to put a different `has_scalar` action on `C^n⟮I, M; 𝕜⟯`
+which is defined as `f • r = f(x) * r`. -/
+@[nolint unused_arguments] def pointed_smooth_map (x : M) := C^n⟮I, M; 𝕜⟯
```

Comment:
Should this be `pointed_times_cont_mdiff_map` since we allow `n < ∞`?

Reply:
It should but the only reason why it is like that is that otherwise I would have to create a different notation, because the token `C^∞⟮` would break the current notation for smooth functions. I'd rather keep `pointed_smooth_map` instead of `pointed_times_cont_mdiff_map` since I believe the case `n \neq ∞` will never be explored. It is acceptable?

Reply:
Do you want to change the name even if the rest of the theory makes sense only for smooth stuff?

Reply:
This pointed version does not have other uses than in this theory to my knowledge.